test/morgan.js

```diff
@@ -1316,6 +1317,102 @@ describe('morgan.compile(format)', function () {
   })
 })
 
+describe('morgan.token(name, function)', function () {
+  it('should overwrite existing functions if called multiple times', function (done) {
+    function token (req, res, arg) {
+      return arg && url.parse(req.url)[arg]
+    }
+    var urlToken = morgan.url
+    morgan.token('url', token)
+    assert.equal(morgan.url, token)
+    morgan.token('url', urlToken)
+    done()
+  })
+
+  describe('should use the string `-` if return value of the token function is falsey', function () {
```

Comment:
The `describe` should be keep below the `it` at the same level, that way they are ordered in the code in the way that mocha outputs it, which makes it so much easier to correlate them from the test output into the file. It's what we try to keep our style as.